src/scheduler/fifo.c

```diff
@@ -798,6 +745,25 @@ scheduling_cycle(int sd, char *jobid)
 	return 0;
 }
 
+static int
+has_prio_cmd(svr_t *sconn)
+{
+	sched_cmd_t pcmd;
+
+	if (sconn->sfd == -1)
+		return 0;
+
+	if (!get_sched_cmd_noblk(sconn->sfd, &pcmd))
+		return 0;
+
+	if (pcmd.cmd == SCH_SCHEDULE_RESTART_CYCLE) {
```

Comment:
What if the command is other than SCH_SCHEDULE_RESTART_CYCLE. If this is the case we actually lose that scheduling command.

Reply:
This is same as earlier, earlier also it was ignoring any other command.

Reply:
No. This is not same as earlier. It is because prior to this change only SCH_SCHEDULE_RESTART_CYCLE used to be sent on secondary connection. Apart from this nothing else was sent. But as per the current design all scheduling commands are sent on the secondary connection and hence there is a possibility that we might end up losing a command other than SCH_SCHEDULE_RESTART_CYCLE. This definitely causes problems going forward especially if we have multiple servers.

Reply:
@suresh-thelkar server doesn't send any other command except restart_cycle till its get end cycle notification. and we read prio command only from server who triggered sched cycle so in case multiple server will not it won't any problem.

Reply:
The command to restart a cycle comes from a mom hook.  If we only read high priority commands from the server which started our cycle, what happens if a node from a different server sends a restart cycle?  Don't we need to read high priority commands from all servers?

Reply:
@bhroam True we will have to read commands from all servers in case of multi server, but that change will come (soon) as part of multi server changes (as there are few dependent code which we need first before we can make change here for multi-server).